### PR TITLE
93285 - remove unneeded aria labels from appointment list

### PR DESCRIPTION
**Pull request by ryanshaw**
## Summary

This PR corrects some aria labelling on the appointment list screens and addresses the feedback that came out of a staging review.

See https://github.com/department-of-veterans-affairs/va.gov-team/issues/55864 and https://github.com/department-of-veterans-affairs/va.gov-team/issues/55851 for the full context.

## Related issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/93285

## Testing done

- Unit tests
- e2e tests
- Manual inspection of elements

## Acceptance criteria

- [ ] For the aria-label on a div in the first column of each appointment:
   - Remove the `aria-label` attribute, and move its content into `<span class="sr-only">` tags inside the div it's currently attached to. (`aria-label` isn't a valid attribute for `div`s)
   - Remove the text ` appointment on` from the content. The span should just contain the date information, i.e.: `<span class="sr-only">Friday, September 20, 9:40 a.m. Pacific Time</span>`

image

---

- [ ] Remove `<span class="sr-only">Appointments in</span>` from the H2 month/year headings:

image

---

- [ ] Remove the `aria-labelledby` attribute from the ul that follows the heading. The context of the heading is enough to create the association. (Also remove the ID on the heading if this isn't being used for other functionality).

image

---

### Quality Assurance & Testing

- [ ] I updated unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed

### Error Handling

- [ ] Browser console contains no warnings or errors.

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

**Review by rmessina1010 — DISMISSED** [review]
LGTM

**Review by JunTaoLuo — COMMENTED** [review]
One quick question but otherwise look good.

File: src/applications/vaos/appointment-list/components/UpcomingAppointmentsList.jsx [hunk 1]
```diff
@@ -101,7 +101,6 @@ export default function UpcomingAppointmentsList() {
               id={`appointment_list_${monthDate.format('YYYY-MM')}`}
```
Comment (JunTaoLuo) [inline src/applications/vaos/appointment-list/components/UpcomingAppointmentsList.jsx:101]: Hmm I think this should be removed as well. Same with `aria-labelledby` on line 108.

**Review by JunTaoLuo — COMMENTED** [review]
I think you missed a spot 😄

File: src/applications/vaos/appointment-list/components/UpcomingAppointmentsList.jsx (hunk 1 above)
Comment (JunTaoLuo) [reply src/applications/vaos/appointment-list/components/UpcomingAppointmentsList.jsx:101]: I meant the id here as well

**Review by simiadebowale — DISMISSED** [review]

**Review by JunTaoLuo — APPROVED** [review]

**Pull request merged**